romancal/outlier_detection/outlier_detection_step.py

```diff
@@ -55,19 +54,46 @@ def process(self, input_models):
 
         self.input_models = input_models
         self.input_container = isinstance(self.input_models, ModelContainer)
+        self.skip = False
+
+        # make sure input is a ModelContainer
+        if self.input_container:
+            return self._run_process()
+
+        # if input is not a ModelContainer, log warning message and raise TypeError
+        self.log.warning(
+            "Input is not a ModelContainer. Outlier detection step will be SKIPPED."
+        )
+        raise TypeError
```

Comment:
I don't think raising an error is desirable in this case. It may be better to issue a warning and skip the step.